### PR TITLE
Run3-gex89 Attempt to define unqiue names of materials everywhere

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Attempt to define unqiue names of materials everywhere. Remove duplicate names from materials.xml (which are defined in tracker.xml) and make sure that names of all materials for PPS to be preceded by PPS_

#### PR validation:

Use some of runTheMatrix test workflows 

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cvuosalo @civanch I have made changes and removed possibility of defining materials with the same name. Checked for materials defined for tracker, calorimeters and forward systems. The relevant cfi's  are in Geometry/VeryForwardData/python (for DDD) and in Geometry/VeryForwardData/dd4hep (for DD4Hep). We can start testing now

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by cvuosalo** [comment]
@bsunanda Have you seen in any preliminary tests that the discrepancies are resolved by this change?

**Comment by srimanob** [comment]
@bsunanda @civanch @cvuosalo 
Thanks very much. 
If I pick up this PR on top of 12_1, is it enough to test with XML geometry used in SIM, RECO steps?

File: Geometry/VeryForwardData/data/dd4hep/cmsExtendedGeometry2021,xml [hunk 1]
```diff
@@ -0,0 +1,377 @@
+<?xml version="1.0"?>
```
Comment (civanch) [inline Geometry/VeryForwardData/data/dd4hep/cmsExtendedGeometry2021,xml:1]: @bsunanda , there is typo in this file name ",xml"

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

This PR does not fix the issue but is useful in general.

**Comment by cvuosalo** [comment]
hold

**Comment by cvuosalo** [comment]
Let's wait a little bit on this PR. It makes a big perturbation to the Geometry materials right at the same time we are checking the materials for other issues.

**Comment by cvuosalo** [comment]
unhold

**Comment by cvuosalo** [comment]
Let's go forward with this PR and get it merged soon.

**Comment by srimanob** [comment]
+Upgrade

**Comment by perrotta** [comment]
> 
> This PR does not fix the issue but is useful in general.

Just for completeness of information: which is the issue that is mentioned several times in this thread, and which is "not fixed" by this PR? @bsunanda @civanch @cvuosalo

**Comment by bsunanda** [comment]
There was a discrepancy of material budget in the tracker between the xml and db version for the dd4hep implementation. This PR tries to avoid duplicate material names which could be a cause. It was later on established that the order in which materials are defined was causing this and this was taken care of in a separate PR. However avoiding duplicate names is a good practice and defining materials with the same name but of different properties is confusing and must be avoided.

**Comment by bsunanda** [comment]
@perrotta I tried to send a comment. It was discussed in the Simulation meetings (and chats) at great length

**Comment by bsunanda** [comment]
@qliphy @perrotta @cvuosalo @civanch Please include this in IB at the earliest. We need to define a new scenario to be used  for the next GT

**Comment by perrotta** [comment]
+1
- Renamed materials whenever it was instrumental to cleanup the configurations in geometry
- (@bsunanda : we had to wait for a clean IB and able to compile with #34972 merged: that's why this PR was not merged during the week end. Following developments, e.g. #34974, can be tested and eventually integrated on top the forhcoming IBs which contain this one)

**Pull request merged**